### PR TITLE
Integrating the relay server with fybrik.

### LINKED ISSUE #65: Use a configuration in implementing Wasm transformations
Adding two parameters to transformation function, one is a pointer to block of memory that stores the configuration and the second is its size. In WASM side, the function can parse the configuration (as Json) and uses the values in it.

### PR DESCRIPTION
**Pull request by Mohammad-nassar10**
Adding a relay producer that requests from flight server to read the data and it can perform transformations using Wasm images stored in OCI registry.

Adding a Wasm module that performs a filtering action according to a given operation and value.

Closes #65 

Signed-off-by: nassarmu10 <mohammad.nassar@ibm.com>

### CODE REVIEW

File: docker/Dockerfile.relay-server2 [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM openjdk:11
+
+RUN apt-get update \
+    && apt-get install -y time
+
+COPY logging.xml /root
+COPY flight-server-all.jar /root/
+COPY run_relay_server2.sh /root/
```
Comment (cdoron) [inline docker/Dockerfile.relay-server2]: It is unclear what the difference is between `run_relay_server.sh` and `run_relay_server2.sh`. Please rename

File: docker/create_docker_images.sh [hunk 2]
```diff
@@ -4,15 +4,17 @@ if [ ! -f logging.xml ]
...
+cp ../flight-client/build/libs/flight-client-all.jar .
 
-docker build -f Dockerfile.client --tag example_client:latest .
-docker build -f Dockerfile.memory-server --tag example_memory:latest .
-docker build -f Dockerfile.relay-server --tag example_relay:latest .
-docker build -f Dockerfile.both-servers --tag example_both:latest .
+# docker build -f Dockerfile.client --tag example_client:latest .
+# docker build -f Dockerfile.memory-server --tag example_memory:latest .
+# docker build -f Dockerfile.relay-server --tag example_relay:latest .
+# docker build -f Dockerfile.both-servers --tag example_both:latest .
```
Comment (cdoron) [inline docker/create_docker_images.sh]: consider removing all commented-out lines, throughout

File: flight-server/src/main/java/org/m4d/adp/flight_server/RelayServer.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        remote_port = Integer.valueOf(line.getOptionValue("remote_port", "12233"));
+        
+        host = line.getOptionValue("host", "0.0.0.0");
+        port = Integer.valueOf(line.getOptionValue("port", "12232"));
+
+        // Read the configuration file as Yaml
+        InputStream inputStream = new FileInputStream(new File("/etc/conf/conf.yaml"));
+        String yamlStr = CharStreams.toString(new InputStreamReader(inputStream, Charsets.UTF_8));
+        ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+        // Convert the Yaml to Json
```
Comment (cdoron) [inline flight-server/src/main/java/org/m4d/adp/flight_server/RelayServer.java:78]: is the conversion from yaml to json still required?

File: wasm_interface/src/allocator.rs [hunk 4]
```diff
@@ -1,11 +1,17 @@
...
+use crate::{oci_cache, types::{WasmModule, Pointer}};
+
 
-// A function that takes a path to `.wasm` file and returns a Pointer of type WasmModule.
+// A function that takes a path to OCI image and returns a Pointer of type WasmModule.
 #[no_mangle]
-pub fn wasmInstance(path: String) -> i64{
-    let wasm_bytes_file = std::fs::read(path).unwrap();
+pub fn wasmInstance(path: String) -> i64 {
+    println!("gg");
```
Comment (cdoron) [inline wasm_interface/src/allocator.rs]: required?

**Review by cdoron — COMMENTED** [review]
As we've discussed, I think it would be best to copy the contents of
https://github.com/fybrik/arrow-data-plane/pull/61
to a documentation file within the repository

**Review by the-mesh-for-data — APPROVED** [review]

File: flight-client/src/main/java/org/m4d/adp/flight_client/MyFlightClient.java [hunk 5]
```diff
@@ -44,13 +46,33 @@ public static void main(String[] args) {
...
             // System.out.println(i);
-            // System.out.println("client");
+            // // System.out.println("client");
             // System.out.println(root.getVector(0));
             // System.out.println(root.getVector(1));
             // System.out.println(root.getVector(2));
             // System.out.println(root.getVector(3));
         }
         long finish = System.currentTimeMillis();
+        ////
```
Comment (cdoron) [inline flight-client/src/main/java/org/m4d/adp/flight_client/MyFlightClient.java]: What does this section (surrounded by '////') signify?

File: flight-server/src/main/java/org/m4d/adp/flight_server/RelayServer.java (hunk 3 above)
Comment (cdoron) [reply flight-server/src/main/java/org/m4d/adp/flight_server/RelayServer.java:78]: Let's discuss this conversion. Thanks

**Review by cdoron — APPROVED** [review]

**Pull request merged**